src/cluster_asm.c

```diff
@@ -524,73 +549,64 @@ static void clusterMigrationCommandImport(client *c) {
     slotRangeArray *slot_ranges = parseSlotRangesOrReply(c, c->argc, 3);
     if (!slot_ranges) return;
 
+    sds task_id = NULL;
     sds err = NULL;
-    if (asmCreateImportTask(slot_ranges, &err) != C_OK) {
+    if (clusterAsmProcess(&task_id, ASM_EVENT_IMPORT_START, slot_ranges, &err) != C_OK) {
         addReplyErrorSds(c, err);
         return;
     }
 
-    addReply(c, shared.ok);
-}
-
-/* Cancel atomic slot migration tasks that overlap with the given slot
- * range. Returns the number of cancelled tasks. */
-static int cancelTasksForSlotRange(slotRange *req_range) {
-    int num_cancelled = 0;
-    listIter li;
-    listNode *ln;
-
-    listRewind(asmManager->tasks, &li);
-    while ((ln = listNext(&li)) != NULL) {
-        asmTask *task = listNodeValue(ln);
-
-        if (task->state == ASM_DONE)
-            continue;
-
-        /* Cancel the task if it overlaps with the slot ranges of the task. */
-        if (slotRangeArrayOverlaps(task->slot_ranges, req_range)) {
-            asmTaskSetFailed(task, "Cancelled by user");
-            task->state = ASM_CANCELED;
-            asmTaskComplete(task);
-            num_cancelled++;
-            continue;
-        }
-    }
-    return num_cancelled;
+    addReplyBulkCString(c, task_id);
 }
 
-/* CLUSTER MIGRATION CANCEL <start-slot end-slot [start-slot end-slot ...]>
+/* CLUSTER MIGRATION CANCEL [ID <id> | ALL]
```

Comment:
if we switch to IDs, I guess it does not make sense to cancel tasks by slot ranges